File: java/src/main/java/ai/bizx/chess/ChessStrategy.java
package ai.bizx.chess;

import java.util.HashMap;
import java.util.Map;

public final class ChessStrategy {
    private long turn;
    private final Map<String, Long> resources = new HashMap<>(Map.of("gold",100L,"food",100L,"science",0L));
    public long strategyTick(String resource, long amount) { resources.merge(resource, amount, Long::sum); return ++turn; }
    public Map<String,Object> uci(String fen, String command) {
        return Map.of("protocol","UCI","fen",fen,"command",command == null ? "go movetime 100" : command,"adapter","external-engine");
    }
    public long turn() { return turn; }
    public Map<String,Long> resources() { return Map.copyOf(resources); }
}

File: rust/src/chess_strategy.rs
use std::collections::HashMap;

#[derive(Default)]
pub struct ChessStrategy { pub turn: u64, pub resources: HashMap<String,i64> }
impl ChessStrategy {
    pub fn new() -> Self { Self { turn: 0, resources: HashMap::from([("gold".into(),100),("food".into(),100),("science".into(),0)]) } }
    pub fn tick(&mut self, resource: &str, amount: i64) -> u64 { *self.resources.entry(resource.into()).or_default() += amount; self.turn += 1; self.turn }
    pub fn uci(&self, fen: &str, command: &str) -> (String,String,String) { ("UCI".into(), fen.into(), if command.is_empty() { "go movetime 100".into() } else { command.into() }) }
}
